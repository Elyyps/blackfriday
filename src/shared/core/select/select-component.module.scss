.select {
  padding: 3px;
  display: flex;
  flex-direction: row;
  align-content: center;
  align-items: center;
  position: relative;
  select {
    border: 0 !important; /*Removes border*/
    -webkit-appearance: none; /*Removes default chrome and safari style*/
    -moz-appearance: none; /* Removes Default Firefox style*/
    background-position: 82px 7px; /*Position of the background-image*/
    text-indent: 0.01px; /* Removes default arrow from firefox*/
    text-overflow: ""; /*Removes default arrow from firefox*/
    background: transparent;
    font-size: 15px;
    /*My custom style for fonts*/
  }
  &-icon {
    margin-top: 3px;
    position: absolute;
    top: 4px;
    right: -5px;
  }
}
