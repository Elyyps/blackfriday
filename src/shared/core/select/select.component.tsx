import * as React from "react";
import styles from "./select-component.module.scss";
import ArrowLongDown from "@assets/icons/arrow-long-down.svg";
import { IconComponent } from "../icon";

export interface ISelectComponentProps {
  icon?: string;
  iconSize?: string;
  onSelect: (value: string) => void;
  options: string[];
}

const SelectComponent = (props: ISelectComponentProps) => {
  const handelChange = (event: any) => {
    const newValue = event.target.value;
    props.onSelect(newValue);
  };

  return (
    <div className={styles["select"]}>
      <select className={styles["select-title"]} onChange={handelChange} onSelect={() => props.onSelect}>
        {props.options.map(value => (
          <option aria-selected value={value}>
            {value}
          </option>
        ))}
      </select>
      <span className={` ${styles["select-icon"]} `}>
        <IconComponent icon={props.icon || ArrowLongDown} size={props.iconSize || "12px"} />
      </span>
    </div>
  );
};

export { SelectComponent };
