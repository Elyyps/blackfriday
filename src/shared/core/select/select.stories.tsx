import * as React from "react";

import { storiesOf } from "@storybook/react";
import { withA11y } from "@storybook/addon-a11y";
import { withProvider } from "@app/util";
import { SelectComponent } from "./select.component";
const options = ["newest", "price", "relevance"];

storiesOf("Select", module)
  .addDecorator(withA11y)
  .addDecorator(withProvider)
  .add("Basic implementation", () => <SelectComponent options={options} onSelect={() => ""} />);
