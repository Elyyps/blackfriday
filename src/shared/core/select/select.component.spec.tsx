
import React from "react";
import { shallow } from "enzyme";
import { SelectComponent } from "./select.component";

describe("[Select]", () => {
  it("should render component without crashing", () => {
    const renderedComponent = shallow(
      <SelectComponent />
    );
    expect(renderedComponent).toMatchSnapshot();
  });
});

  