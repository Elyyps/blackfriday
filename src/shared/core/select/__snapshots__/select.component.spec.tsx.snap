// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Select] should render component without crashing 1`] = `
<div
  className="select"
>
  <select
    aria-label="Select"
    className="select-title"
    onChange={[Function]}
    onSelect={[Function]}
  >
    <option
      aria-selected={true}
      key="0"
      value="newest"
    >
      newest
    </option>
    <option
      aria-selected={true}
      key="1"
      value="price"
    >
      price
    </option>
    <option
      aria-selected={true}
      key="2"
      value="relevance"
    >
      relevance
    </option>
  </select>
  <span
    className="select-icon "
  >
    <IconComponent
      icon="arrow-long-down.svg"
      size="12px"
    />
  </span>
</div>
`;
